openmdao/core/implicitcomponent.py

```diff
@@ -31,6 +32,11 @@ def __init__(self, **kwargs):
         """
         super().__init__(**kwargs)
 
+        if 'tags' not in kwargs:
```

Comment:
This doesn't seem like it would actually do anything.  'tags' isn't passed to component __init__ methods except in the case of IndepVarComp which allows it because the user can add an output as part of the __init__ args.  I think you can get rid of this whole block.  You're already taking care of the 'allow_desvar' tag in the add_output method.